### PR TITLE
sl-9: jwt

### PR DESCRIPTION
**Pull request by vbazhynov**
Sign in response 
image
Sign Up Response
image
Token in the storage 
image

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   },
   "devDependencies": {
     "@types/convict": "6.1.1",
+    "@types/jsonwebtoken": "9.0.1",
```
Comment (what1s1ove) [inline backend/package.json]:
```suggestion
```
image

File: backend/src/packages/auth/auth.service.ts [hunk 2]
```diff
@@ -1,25 +1,32 @@
 import {
   type UserSignInRequestDto,
+  type UserSignInResponseDto,
   type UserSignUpRequestDto,
   type UserSignUpResponseDto,
 } from '~/packages/users/libs/types/types.js';
 import { type UserService } from '~/packages/users/user.service.js';
 
+import { createToken } from './libs/helpers/token/create-token/create.token.helper.js';
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
```
create service for it

File: backend/src/packages/auth/auth.service.ts [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
   private userService: UserService;
 
   public constructor(userService: UserService) {
     this.userService = userService;
   }
 
-  public signIn(userRequestDto: UserSignInRequestDto): UserSignInRequestDto {
-    return userRequestDto;
+  public signIn(userRequestDto: UserSignInRequestDto): UserSignInResponseDto {
+    const id = Math.floor(Math.random() * 100); //replase with proper id from database
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]: you should to do it here

File: backend/src/packages/auth/libs/helpers/token/create-token/create.token.helper.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import jwt from 'jsonwebtoken';
+
+import { config } from '~/libs/packages/config/config.js';
+
+const createToken = (data: number): string => {
+  return jwt.sign({ data }, config.ENV.JWT.SECRET_KEY, {
+    expiresIn: '24h',
```
Comment (what1s1ove) [inline backend/src/packages/auth/libs/helpers/token/create-token/create.token.helper.ts]: add this to `config`

File: frontend/src/slices/auth/auth.slice.ts [hunk 5]
```diff
@@ -21,8 +25,10 @@ const { reducer, actions, name } = createSlice({
     builder.addCase(signUp.pending, (state) => {
       state.dataStatus = DataStatus.PENDING;
     });
-    builder.addCase(signUp.fulfilled, (state) => {
+    builder.addCase(signUp.fulfilled, (state, action) => {
       state.dataStatus = DataStatus.FULFILLED;
+      void storage.set(StorageKey.TOKEN, action.payload.token);
```
Comment (what1s1ove) [inline frontend/src/slices/auth/auth.slice.ts]:
```suggestion
```
you should to do it in action

File: frontend/src/slices/auth/auth.slice.ts [hunk 6]
```diff
@@ -1,16 +1,20 @@
...
 import { signUp } from './actions.js';
 
 type State = {
   dataStatus: ValueOf<typeof DataStatus>;
+  user: UserSignUpResponseDto;
 };
 
 const initialState: State = {
   dataStatus: DataStatus.IDLE,
+  user: { token: '', user: { id: 1, email: '' } },
```
Comment (what1s1ove) [inline frontend/src/slices/auth/auth.slice.ts]:
```suggestion
  user: null,
```
default should be null

File: shared/src/packages/users/libs/types/user-sign-in-response-dto.type.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+type UserSignInResponseDto = {
+  token: string;
+  user: { id: number; email: string };
```
Comment (what1s1ove) [inline shared/src/packages/users/libs/types/user-sign-in-response-dto.type.ts]:
```suggestion
  user: {
    id: number;
    email: string
    };
```
fix for all

Comment (what1s1ove) [reply shared/src/packages/users/libs/types/user-sign-in-response-dto.type.ts]: create a type for user

something like `UserAutResponse`

File: backend/src/packages/auth/auth.service.ts (hunk 3 above)
Comment (vbazhynov) [reply backend/src/packages/auth/auth.service.ts]: You mean implement user find in repositiry?

File: backend/src/packages/auth/libs/helpers/token/create-token/create.token.helper.ts (hunk 4 above)
Comment (vbazhynov) [reply backend/src/packages/auth/libs/helpers/token/create-token/create.token.helper.ts]: Yoy mean expiration?

**Review by what1s1ove — CHANGES_REQUESTED** [review]
don't forget to change swagger doc

File: backend/package.json (hunk 1 above)
Comment (what1s1ove) [inline backend/package.json]:
```suggestion
```

File: backend/src/libs/interfaces/repository.interface.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 interface IRepository<T = unknown> {
-  find(): Promise<T>;
+  find(payload: unknown): Promise<T>;
```
Comment (what1s1ove) [inline backend/src/libs/interfaces/repository.interface.ts]:
```suggestion
  find(id: unknown): Promise<T>;
```

File: backend/src/libs/interfaces/service.interface.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 interface IService<T = unknown> {
-  find(): Promise<T>;
+  find(payload: unknown): Promise<T>;
```
Comment (what1s1ove) [inline backend/src/libs/interfaces/service.interface.ts]:
```suggestion
  find(id: unknown): Promise<T>;
```

File: backend/src/libs/packages/config/config.package.ts [hunk 10]
```diff
@@ -69,6 +69,26 @@ class Config implements IConfig {
...
+          format: String,
+          env: 'ALGORITHM',
+          default: null,
+        },
+        EXP_TIME: {
+          doc: 'Token expiration time',
+          format: String,
+          env: 'EXP_TIME',
+          default: null,
+        },
```
Comment (what1s1ove) [inline backend/src/libs/packages/config/config.package.ts]:
```suggestion
```

```
    this.ENV = this.envSchema.getProperties();
    this.auth = this.authConfig
    this.logger.info('.env file found and successfully parsed!');
```
move these two variables to separate config

File: backend/src/packages/auth/auth.service.ts [hunk 11]
```diff
@@ -1,5 +1,10 @@
+import * as jose from 'jose';
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
```
create a token package for it

avoid `* as` import

File: backend/src/packages/users/user.repository.ts [hunk 12]
```diff
@@ -11,6 +11,14 @@ class UserRepository
     this.userModel = userModel;
   }
 
+  public async find(email: string): ReturnType<IRepository['find']> {
```
Comment (what1s1ove) [inline backend/src/packages/users/user.repository.ts]: `find` should work with `id`
create `findByEmail` method for it

Comment (what1s1ove) [inline backend/src/packages/users/user.repository.ts]:
```suggestion
  public async find(email: string): Entity {
```
should be user entity here

File: backend/src/packages/users/user.service.ts [hunk 13]
```diff
@@ -15,6 +15,10 @@ class UserService implements Omit<IService, 'find' | 'update' | 'delete'> {
     this.userRepository = userRepository;
   }
 
+  public async find(payload: string): ReturnType<IService['find']> {
```
Comment (what1s1ove) [inline backend/src/packages/users/user.service.ts]: create a popriet return type

File: frontend/src/slices/auth/actions.ts [hunk 14]
```diff
@@ -1,5 +1,6 @@
 import { createAsyncThunk } from '@reduxjs/toolkit';
 
+import { storage, StorageKey } from '~/libs/packages/storage/storage.js';
```
Comment (what1s1ove) [inline frontend/src/slices/auth/actions.ts]:
```suggestion
import { StorageKey } from '~/libs/packages/storage/storage.js';
```
all services should be passed through extra arguments

File: shared/src/packages/users/libs/types/user-sign-up-response-dto.type.ts [hunk 15]
```diff
@@ -1,8 +1,8 @@
+import { type UserRegisterResponse } from './user-register-response.type.js';
+
 type UserSignUpResponseDto = {
-  id: number;
-  email: string;
-  firstName: string;
-  lastName: string;
+  token: string;
+  user: UserRegisterResponse;
```
Comment (what1s1ove) [inline shared/src/packages/users/libs/types/user-sign-up-response-dto.type.ts]:
```suggestion
  user: UserAuthResponse;
```
for sign-in and sign-up should be the same type for the user

**Review by what1s1ove — DISMISSED** [review]

File: backend/src/packages/users/user.service.ts [hunk 16]
```diff
@@ -3,8 +3,8 @@ import { UserEntity } from '~/packages/users/user.entity.js';
 import { type UserRepository } from '~/packages/users/user.repository.js';
 
 import {
-  type UserAuthResponse,
   type UserGetAllResponseDto,
+  type UserRegisterResponse,
```
Comment (kuslosolodka) [inline backend/src/packages/users/user.service.ts]: maybe you should name this type as UserSignUpResponse

File: frontend/src/slices/auth/actions.ts [hunk 17]
```diff
@@ -12,10 +13,11 @@ const signUp = createAsyncThunk<
   UserSignUpResponseDto,
   UserSignUpRequestDto,
   AsyncThunkConfig
->(`${sliceName}/sign-up`, (registerPayload, { extra }) => {
+>(`${sliceName}/sign-up`, async (registerPayload, { extra }) => {
```
Comment (kuslosolodka) [inline frontend/src/slices/auth/actions.ts]: maybe you should name registerPayload as signUpPayload

File: backend/src/packages/auth/auth.service.ts (hunk 11 above)
Comment (vbazhynov) [reply backend/src/packages/auth/auth.service.ts]: What do you mean with "create package"? trying to find something about it half of night and still don`t have understanding what to do.

File: backend/src/libs/interfaces/repository.interface.ts [hunk 18]
```diff
@@ -1,5 +1,6 @@
 interface IRepository<T = unknown> {
-  find(payload: unknown): Promise<T>;
+  find(id: number): Promise<T>;
```
Comment (kuslosolodka) [inline backend/src/libs/interfaces/repository.interface.ts:2]: i think it should be id:unknown, email: unknown, check create method payload parameter

File: backend/src/libs/interfaces/repository.interface.ts [hunk 19]
```diff
@@ -1,5 +1,6 @@
...
+  findByEmail(email: string): Promise<T>;
```
Comment (kuslosolodka) [inline backend/src/libs/interfaces/repository.interface.ts]: i think you should use id: unknown and email:unknown, check create method

File: backend/src/libs/interfaces/service.interface.ts [hunk 20]
```diff
@@ -1,5 +1,6 @@
 interface IService<T = unknown> {
-  find(payload: unknown): Promise<T>;
+  find(id: number): Promise<T>;
+  findByEmail(email: string): Promise<T>;
```
Comment (kuslosolodka) [inline backend/src/libs/interfaces/service.interface.ts]: i think you should use id: unknown and email: unknown , same as  in repository interface

File: backend/src/packages/auth/auth.service.ts (hunk 11 above)
Comment (vbazhynov) [reply backend/src/packages/auth/auth.service.ts]: I thing i got it. No will try to fix

File: backend/src/libs/interfaces/repository.interface.ts (hunk 19 above)
Comment (dimapopovych) [reply backend/src/libs/interfaces/repository.interface.ts]: First of all you shouldn't add findByEmail into IRepository. About id I think it's fine all id in db are numbers but we will see

File: backend/src/libs/interfaces/service.interface.ts (hunk 20 above)
Comment (vbazhynov) [reply backend/src/libs/interfaces/service.interface.ts]: Why? we know types of email and ID

File: backend/src/libs/interfaces/repository.interface.ts [hunk 21]
```diff
@@ -1,5 +1,6 @@
 interface IRepository<T = unknown> {
-  find(): Promise<T>;
+  find(id: number): Promise<T>;
+  findByEmail(email: string): Promise<T>;
```
Comment (what1s1ove) [inline backend/src/libs/interfaces/repository.interface.ts]: agreed. you don't need them at all
```suggestion
```

File: backend/src/libs/interfaces/repository.interface.ts [hunk 22]
```diff
@@ -1,5 +1,6 @@
 interface IRepository<T = unknown> {
-  find(): Promise<T>;
+  find(id: number): Promise<T>;
```
Comment (what1s1ove) [inline backend/src/libs/interfaces/repository.interface.ts]:
```suggestion
  find(id: unknown): Promise<T>;
```

File: backend/src/libs/interfaces/service.interface.ts [hunk 23]
```diff
@@ -1,9 +1,10 @@
 interface IService<T = unknown> {
-  find(): Promise<T>;
+  find(id: number): Promise<T>;
   findAll(): Promise<{
     items: T[];
   }>;
   create(payload: unknown): Promise<T>;
+  findByEmail(email: string): Promise<T>;
```
Comment (what1s1ove) [inline backend/src/libs/interfaces/service.interface.ts]:
```suggestion
  find(id: unknown): Promise<T>;
  findAll(): Promise<{
    items: T[];
  }>;
  create(payload: unknown): Promise<T>;
```

File: backend/src/libs/packages/config/config.package.ts [hunk 24]
```diff
@@ -11,7 +11,6 @@ class Config implements IConfig {
   private logger: ILogger;
 
   public ENV: EnvironmentSchema;
```
Comment (what1s1ove) [inline backend/src/libs/packages/config/config.package.ts:13]:
```suggestion
  public ENV: EnvironmentSchema;

```

File: backend/src/libs/packages/config/config.ts [hunk 25]
```diff
@@ -1,8 +1,11 @@
 import { logger } from '~/libs/packages/logger/logger.js';
 
 import { Config } from './config.package.js';
+import { TokenConfig } from './token.config.package.js';
 
 const config = new Config(logger);
+const tokenConfig = new TokenConfig(logger);
```
Comment (what1s1ove) [inline backend/src/libs/packages/config/config.ts]:
```suggestion
```
you don't need an additional class. just add a new getter `AUTH` to the config class. without ENV parsing for them. just hardcode them

File: backend/src/libs/packages/config/token.config.package.ts [hunk 26]
```diff
@@ -0,0 +1,46 @@
+import convict, { type Config as TConfig } from 'convict';
```
Comment (what1s1ove) [inline backend/src/libs/packages/config/token.config.package.ts]: you don't need an additional class. just add a new getter `AUTH` to the config class. without ENV parsing for them. just hardcode them

File: backend/src/libs/packages/token/libs/token.interface.ts [hunk 27]
```diff
@@ -0,0 +1,4 @@
+interface IToken {
+  createToken(id: number): Promise<string>;
```
Comment (what1s1ove) [inline backend/src/libs/packages/token/libs/token.interface.ts]:
```suggestion
  create(payload: unknown): Promise<string>;
  verify
  decode
```

File: backend/src/libs/packages/token/token-package.ts [hunk 28]
```diff
@@ -0,0 +1,24 @@
+import { SignJWT } from 'jose';
```
Comment (what1s1ove) [inline backend/src/libs/packages/token/token-package.ts]: rename to `backend/src/libs/packages/token/token.package.ts`

File: backend/src/packages/auth/auth.service.ts [hunk 29]
```diff
@@ -1,26 +1,49 @@
...
   type UserSignUpRequestDto,
   type UserSignUpResponseDto,
 } from '~/packages/users/libs/types/types.js';
 import { type UserService } from '~/packages/users/user.service.js';
 
 class AuthService {
   private userService: UserService;
-
-  public constructor(userService: UserService) {
+  private token: Token;
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
  private tokenService: IToken;
```

File: backend/src/packages/auth/auth.service.ts [hunk 30]
```diff
@@ -1,26 +1,49 @@
+import { type Token } from '~/libs/packages/token/token-package.js';
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
import { type Token } from '~/libs/packages/token/token.js';
```

File: backend/src/packages/auth/auth.service.ts [hunk 31]
```diff
@@ -1,26 +1,49 @@
...
 import {
+  type UserAuthResponse,
   type UserSignInRequestDto,
+  type UserSignInResponseDto,
   type UserSignUpRequestDto,
   type UserSignUpResponseDto,
 } from '~/packages/users/libs/types/types.js';
 import { type UserService } from '~/packages/users/user.service.js';
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
type UserService
} from '~/packages/users/users.js;;
```

File: backend/src/packages/users/user.repository.ts [hunk 32]
```diff
@@ -11,6 +11,48 @@ class UserRepository
...
+  public async find(id: number): Promise<UserEntity> {
+    const user = await this.userModel
+      .query()
+      .where('id', id)
+      .first()
+      .withGraphJoined('userDetails');
+
+    if (!user) {
+      throw new Error('User Not Exist');
+    }
```
Comment (what1s1ove) [inline backend/src/packages/users/user.repository.ts]:
```suggestion
    if (!user) {
      return null
    }
```

File: backend/src/packages/users/user.repository.ts [hunk 33]
```diff
@@ -11,6 +11,48 @@ class UserRepository
...
+  public async findByEmail(email: string): Promise<UserEntity> {
+    const user = await this.userModel
+      .query()
+      .where('email', email)
+      .first()
+      .withGraphJoined('userDetails');
+
+    if (!user) {
+      throw new Error('User Not Exist');
+    }
```
Comment (what1s1ove) [inline backend/src/packages/users/user.repository.ts]:
```suggestion
    if (!user) {
       retrun null
    }
```

File: frontend/src/libs/packages/store/store.package.ts [hunk 34]
```diff
@@ -12,6 +12,8 @@ import { userApi } from '~/packages/users/users.js';
 import { reducer as authReducer } from '~/slices/auth/auth.js';
 import { reducer as usersReducer } from '~/slices/users/users.js';
 
+import { storage } from '../storage/storage.js';
```
Comment (what1s1ove) [inline frontend/src/libs/packages/store/store.package.ts]:
```suggestion
import { storage } from '~/libs/packages/storage/storage.js';
```

File: frontend/src/slices/auth/actions.ts [hunk 35]
```diff
@@ -12,10 +13,11 @@ const signUp = createAsyncThunk<
   UserSignUpResponseDto,
   UserSignUpRequestDto,
   AsyncThunkConfig
->(`${sliceName}/sign-up`, (registerPayload, { extra }) => {
-  const { authApi } = extra;
-
-  return authApi.signUp(registerPayload);
+>(`${sliceName}/sign-up`, async (signUpPayload, { extra }) => {
+  const { authApi, storage } = extra;
+  const user = await authApi.signUp(signUpPayload);
```
Comment (what1s1ove) [inline frontend/src/slices/auth/actions.ts]:
```suggestion
  const { user, tocken } = await authApi.signUp(signUpPayload);
```

File: shared/src/libs/packages/config/libs/interfaces/token.config.interface.ts [hunk 36]
```diff
@@ -0,0 +1,5 @@
+interface ITokenConfig<T> {
```
Comment (what1s1ove) [inline shared/src/libs/packages/config/libs/interfaces/token.config.interface.ts]:
```suggestion
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by vbazhynov — COMMENTED** [review]
All Done

File: backend/src/libs/packages/config/libs/interfaces/config.interface.ts [hunk 37]
```diff
@@ -1,7 +1,7 @@
 import { type IConfig as ILibraryConfig } from 'shared/build/index.js';
 
-import { type EnvironmentSchema } from '../types/types.js';
+import { type AuthConfig, type EnvironmentSchema } from '../types/types.js';
 
-interface IConfig extends ILibraryConfig<EnvironmentSchema> {}
+interface IConfig extends ILibraryConfig<EnvironmentSchema, AuthConfig> {}
```
Comment (what1s1ove) [inline backend/src/libs/packages/config/libs/interfaces/config.interface.ts]:
```suggestion
interface IConfig extends ILibraryConfig<EnvironmentSchema> {
  AUTH: AuthConfig
}
```
you need it only of the backed

File: backend/src/libs/packages/config/config.package.ts [hunk 38]
```diff
@@ -69,6 +70,21 @@ class Config implements IConfig {
...
+        },
+      },
+    });
+  }
+
+  private get authConfig(): TConfig<AuthConfig> {
+    return convict<AuthConfig>({
+      ALGORITHM: 'HS256',
+      EXP_TIME: '24h',
     });
```
Comment (what1s1ove) [inline backend/src/libs/packages/config/config.package.ts]:
```suggestion
    return {
      ALGORITHM: 'HS256',
      EXP_TIME: '24h',
    };
```

File: backend/src/libs/packages/config/config.package.ts [hunk 39]
```diff
@@ -22,8 +21,10 @@ class Config implements IConfig {
       allowed: 'strict',
       output: (message) => this.logger.info(message),
     });
-
+    this.authConfig.load({});
+    this.AUTH = this.authConfig.getProperties();
```
Comment (what1s1ove) [inline backend/src/libs/packages/config/config.package.ts]:
```suggestion
    this.AUTH = this.authConfig;
```

File: backend/src/libs/packages/config/config.package.ts [hunk 40]
```diff
@@ -69,6 +70,21 @@ class Config implements IConfig {
...
+          doc: 'Secret key for token generation',
+          format: String,
+          env: 'SECRET_KEY',
+          default: null,
+        },
+      },
+    });
+  }
+
+  private get authConfig(): TConfig<AuthConfig> {
```
Comment (what1s1ove) [inline backend/src/libs/packages/config/config.package.ts]:
```suggestion
  private get authConfig(): AuthConfig {
```

File: backend/src/packages/projects/project.service.ts [hunk 41]
```diff
@@ -8,7 +8,9 @@ import {
   type ProjectGetAllResponseDto,
 } from './libs/types/types.js';
 
-class ProjectService implements Omit<IService, 'find' | 'update' | 'delete'> {
+class ProjectService
+  implements Omit<IService, 'find' | 'update' | 'delete' | 'findByEmail'>
```
Comment (what1s1ove) [inline backend/src/packages/projects/project.service.ts]:
```suggestion
  implements Omit<IService, 'find' | 'update' | 'delete'>
```

File: backend/src/packages/sites/site.service.ts [hunk 42]
```diff
@@ -8,7 +8,9 @@ import {
   type SiteGetAllResponseDto,
 } from './libs/types/types.js';
 
-class SiteService implements Omit<IService, 'find' | 'update' | 'delete'> {
+class SiteService
+  implements Omit<IService, 'find' | 'update' | 'delete' | 'findByEmail'>
```
Comment (what1s1ove) [inline backend/src/packages/sites/site.service.ts]:
```suggestion
  implements Omit<IService, 'find' | 'update' | 'delete'>
```

File: frontend/src/libs/packages/config/config.package.ts [hunk 43]
```diff
@@ -6,9 +6,10 @@ import { type EnvironmentSchema } from './libs/types/types.js';
 
 class Config implements IConfig {
   public ENV: EnvironmentSchema;
-
+  public AUTH: null;
   public constructor() {
     this.ENV = this.envSchema;
+    this.AUTH = null;
```
Comment (what1s1ove) [inline frontend/src/libs/packages/config/config.package.ts]:
```suggestion
```

File: backend/src/packages/auth/auth.service.ts [hunk 44]
```diff
@@ -1,26 +1,51 @@
+import { type Token } from '~/libs/packages/token/token.js';
 import {
+  type UserAuthResponse,
   type UserSignInRequestDto,
+  type UserSignInResponseDto,
   type UserSignUpRequestDto,
   type UserSignUpResponseDto,
 } from '~/packages/users/libs/types/types.js';
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
  type UserSignInResponseDto,
  type UserSignUpRequestDto,
  type UserSignUpResponseDto,
  type UserService
} from '~/packages/users/users.js';
```
join imports

File: backend/src/packages/auth/auth.service.ts [hunk 45]
```diff
@@ -1,26 +1,51 @@
...
+    userRequestDto: UserSignInRequestDto,
+  ): Promise<UserSignInResponseDto | null> {
+    const { email } = userRequestDto;
+    return await this.login(email);
   }
 
-  public signIn(userRequestDto: UserSignInRequestDto): UserSignInRequestDto {
-    return userRequestDto;
+  private async login(email: string): Promise<UserSignInResponseDto> {
+    const user: UserAuthResponse = (await this.userService.findByEmail(
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]: throw an error if findByEmail return a null

File: backend/src/packages/auth/auth.service.ts [hunk 46]
```diff
@@ -1,26 +1,51 @@
...
+    const user = await this.userService.create(userRequestDto);
+    return {
+      token: await this.createToken(user.id),
+      user,
+    };
   }
+
+  private createToken = async (data: number): Promise<string> => {
+    return await this.tokenServise.create(data);
+  };
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
```
call tokenService directly where you need it

File: backend/src/packages/auth/auth.service.ts [hunk 47]
```diff
@@ -1,26 +1,51 @@
...
   type UserSignUpResponseDto,
 } from '~/packages/users/libs/types/types.js';
-import { type UserService } from '~/packages/users/user.service.js';
+import { type UserService } from '~/packages/users/users.js';
 
 class AuthService {
   private userService: UserService;
-
-  public constructor(userService: UserService) {
+  private tokenServise: Token;
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
  private tokenService: Token;
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/packages/auth/auth.service.ts [hunk 48]
```diff
@@ -1,25 +1,46 @@
...
+    const { email } = userRequestDto;
+    return await this.login(email);
   }
 
-  public signIn(userRequestDto: UserSignInRequestDto): UserSignInRequestDto {
-    return userRequestDto;
+  private async login(email: string): Promise<UserSignInResponseDto> {
+    const user = await this.userService.findByEmail(email);
+    if (!user) {
+      throw new Error('User Not Exist');
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]: throw an application error

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/packages/auth/auth.service.ts [hunk 49]
```diff
@@ -1,25 +1,53 @@
 import {
+  ApplicationError,
+  ApplicationErrorMessage,
+} from 'shared/build/index.js';
```
Comment (andanf-e) [inline backend/src/packages/auth/auth.service.ts]: Do not use direct import from `shared`, these should be imported from enums I believe

File: shared/src/libs/enums/application-error-messages-enum.ts [hunk 50]
```diff
@@ -0,0 +1,5 @@
+const ApplicationErrorMessage = {
+  userNotFound: 'User Not Exist',
```
Comment (andanf-e) [inline shared/src/libs/enums/application-error-messages-enum.ts]: Check **B10**

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: backend/src/packages/auth/auth.service.ts [hunk 51]
```diff
@@ -1,25 +1,51 @@
...
+    return await this.login(email);
   }
 
-  public signIn(userRequestDto: UserSignInRequestDto): UserSignInRequestDto {
-    return userRequestDto;
+  private async login(email: string): Promise<UserSignInResponseDto> {
+    const user = await this.userService.findByEmail(email);
+    if (!user) {
+      throw new ApplicationError({
+        message: ExceptionMessages.userNotFound,
```
Comment (what1s1ove) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
        message: 'your message',
```
do not need an enum for now

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: shared/src/libs/enums/exception-messages-enum.ts [hunk 52]
```diff
@@ -0,0 +1,5 @@
+const ExceptionMessages = {
+  userNotFound: 'User Not Exist',
+};
```
Comment (andanf-e) [inline shared/src/libs/enums/exception-messages-enum.ts]: **B12** + enum names should be in singular form
```suggestion
const ExceptionMessage = {
  USER_NOT_FOUND: 'User Not Exist',
} as const;
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: shared/src/libs/enums/exception-messages-enum.ts [hunk 53]
```diff
@@ -0,0 +1,5 @@
+const ExceptionMessages = {
```
Comment (what1s1ove) [inline shared/src/libs/enums/exception-messages-enum.ts]: remvove it

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Review by andanf-e — APPROVED** [review]

**Pull request merged**